### PR TITLE
add a django settings option to register hstore or not

### PR DESCRIPTION
**Pull request by aleccool213**
this makes project builds cleaner as most django projects that use hstore, install it in the first migration

### CODE REVIEW

File: django_hstore/apps.py [hunk 1]
```diff
@@ -8,8 +8,10 @@
 from psycopg2.extras import register_hstore
 
 
-HSTORE_REGISTER_GLOBALLY = getattr(settings, "DJANGO_HSTORE_ADAPTER_REGISTRATION", "global") == "global"
-CONNECTION_CREATED_SIGNAL_WEAKREF = bool(getattr(settings, "DJANGO_HSTORE_ADAPTER_SIGNAL_WEAKREF", False))
+HSTORE_REGISTER_GLOBALLY = getattr(
+    settings, "DJANGO_HSTORE_ADAPTER_REGISTRATION", "global") == "global"
```
Comment (elliottsj) [inline django_hstore/apps.py]: put one arg per line, with trailing commas

File: django_hstore/apps.py [hunk 2]
```diff
@@ -8,8 +8,10 @@
...
+CONNECTION_CREATED_SIGNAL_WEAKREF = bool(
+    getattr(settings, "DJANGO_HSTORE_ADAPTER_SIGNAL_WEAKREF", False))
```
Comment (elliottsj) [inline django_hstore/apps.py]: same here

File: django_hstore/apps.py [hunk 3]
```diff
@@ -59,8 +61,9 @@ def attach_handler(self, func, vendor=None, unique=False):
 def register_hstore_handler(connection, **kwargs):
     # do not register hstore if DB is not postgres
     # do not register if HAS_HSTORE flag is set to false
-    if connection.vendor != 'postgresql' or \
-       connection.settings_dict.get('HAS_HSTORE', True) is False:
+    if (connection.vendor != 'postgresql' or
+        connection.settings_dict.get('HAS_HSTORE', True) is False or
+        settings.HAS_STORE == True):
```
Comment (elliottsj) [inline django_hstore/apps.py]: Existing comment reads
```py
# do not register if HAS_HSTORE flag is set to false
```

So we're also avoiding registering hstore if `settings.HAS_STORE == True`? What's the difference between `connection.settings_dict` and `settings`?

Comment (aleccool213) [inline django_hstore/apps.py]: ill modify comment but connection.settings_dict is the dict inside the `DATABASES` var in the django settings file

File: django_hstore/apps.py [hunk 4]
```diff
@@ -58,9 +69,13 @@ def attach_handler(self, func, vendor=None, unique=False):
 
 def register_hstore_handler(connection, **kwargs):
     # do not register hstore if DB is not postgres
-    # do not register if HAS_HSTORE flag is set to false
-    if connection.vendor != 'postgresql' or \
-       connection.settings_dict.get('HAS_HSTORE', True) is False:
+    # do not register if HAS_HSTORE in database settings is set to false
+    # do not register if HAS_HSTORE is true in django settings
```
Comment (elliottsj) [inline django_hstore/apps.py]: Seems a bit unintuitive this way. Is there a place to define settings specific to django-hstore? I'd prefer a setting like `DJANGO_HSTORE_REGISTER_HSTORE = False`

Comment (aleccool213) [inline django_hstore/apps.py]: most settings for packages are just in the django settings file, but i do prefer your var name over mine

Comment (aleccool213) [inline django_hstore/apps.py]: @elliottsj updated

**Comment by elliottsj** [comment]
lgtm 🎿

**Pull request merged**